File: DSA-js/slow&fast-pointer/leet19.js
// this question on about the first solve iteration of the finding the 2nd node from teh end of the linkedList
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
var removeNthFromEnd = function(head, n) {
    let senitialNode = new ListNode();
     senitialNode.next = head;
    let prev = senitialNode;
    let curr = head;
    let index = 0
    while(curr)
    {
        curr= curr.next;
        index++;
    }
    let prevlength = index-n;
    for(let i =0;i<prevlength;i++)
    {
        prev = prev.next;
    }
    prev.next=prev.next.next;
    return senitialNode.next;
};

File: DSA-python/slowandfast-pointer/leet19.py

# this question on about the first solve iteration of the finding the 2nd node from teh end of the linkedList
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def removeNthFromEnd(self, head, n):
        """
        :type head: Optional[ListNode]
        :type n: int
        :rtype: Optional[ListNode]
        """
        sentinal = ListNode()
        first = sentinal
        sentinal.next = head
        for i in range(n):
            first = first.next
        second = sentinal
        while first.next:
            first = first.next
            second = second.next
        second.next = second.next.next
        return sentinal.next
        